### PR TITLE
Add rate limiting for API endpoints

### PR DESCRIPTION
**Pull request by aron**
This adds some basic rate limiting in production for the `/api/predictions/[id]` endpoint. We allow a sliding window of 20 requests within 10 seconds to fetch prediction metadata.

If rate limits are hit the client will wait for 10 seconds for the window to ease up before retrying the request. This may have the result of the query experience being slower but we need to keep requests reasonable until we can move to a client side implementation.

We also attempt to reduce the load by introducing a backoff for the pings. Instead of pinging every 500ms indefinitely we now slowly increase the time between requests. Looking at the distribution of traffic. Most requests take between 1 & 3 seconds so this shouldn't have a noticable effect. But for longer predictions >10 seconds we'll greatly reduce the number of requests made.

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -151,17 +153,36 @@ export default function Home({ baseUrl, submissionPredictions }) {
...
+      }
+
+      await sleep(delay + jitter);
       const response = await fetch("/api/predictions/" + prediction.id);
-      prediction = await response.json();
-      console.log(prediction);
+
+      // Handle Rate Limiting
+      if (response.status === 429) {
+        const reset = response.headers.get('X-Ratelimit-Reset') ?? Date.now() + 10_000;
```
Comment (zeke) [inline pages/index.js:176]: TIL [numeric separators](https://github.com/tc39/proposal-numeric-separator) are a thing in JavaScript!?

Comment (zeke) [reply pages/index.js:176]: What do you think about using something like [`ms`](https://github.com/vercel/ms) to make these magic numbers human-friendly?

**Review by zeke — APPROVED** [review]
Thanks for adding this, @aron!

Is there a way to easily test this? Or should we just ship it and see what happens?

**Comment by cbh123** [comment]
Gave it a quick try in the deployment preview and looks good! thank you!

**Pull request merged**